frontend/src/component/segments/EditSegment/EditSegment.tsx

```diff
@@ -78,28 +72,42 @@ export const EditSegment = ({ modal }: IEditSegmentProps) => {
 --data-raw '${JSON.stringify(getSegmentPayload(), undefined, 2)}'`;
     };
 
+    const { highestPermissionChangeRequestEnv } = useChangeRequestsEnabled(
+        segment?.project || ''
+    );
+    const changeRequestEnv = highestPermissionChangeRequestEnv();
+    const activateSegmentChangeRequests =
+        uiConfig?.flags?.segmentChangeRequests && changeRequestEnv;
+    const { addChange } = useChangeRequestApi();
+
     const handleSubmit = async (e: React.FormEvent) => {
         if (segment) {
             e.preventDefault();
             clearErrors();
             try {
                 if (activateSegmentChangeRequests) {
-                    throw new Error(
-                        "You can't add segments to change requests just yet."
-                    );
+                    await addChange(segment.project || '', changeRequestEnv, {
+                        action: 'updateSegment',
+                        feature: null,
+                        payload: { id: segment.id, ...getSegmentPayload() },
+                    });
                 } else {
                     await updateSegment(segment.id, getSegmentPayload());
-                    refetchSegments();
-                    if (projectId) {
-                        navigate(`/projects/${projectId}/settings/segments/`);
-                    } else {
-                        navigate('/segments/');
-                    }
-                    setToastData({
-                        title: 'Segment updated',
-                        type: 'success',
-                    });
                 }
+                refetchSegments();
+                if (projectId) {
```

Comment:
I think `projectId` often is `null` or `undefined` here, even if `segment.project` isn't. Why is that? Shouldn't they be the same? Or is it only set if you navigate there from the projects screen, maybe?

Reply:
because it's read from the hook that has optional query param in the name. Query params are not guaranteed to be there.